torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -475,10 +475,11 @@ void ScriptModuleSerializer::writeArchive(
       [&](const at::Tensor& tensor) {
         // returns a string to use in picker.cpp as storage obj key
         if (tensor_cdata_naming_scheme) {
-          tensor_names.push_back(
+          std::string string_id =
               std::to_string(reinterpret_cast<std::intptr_t>(
-                  tensor.storage().unsafeGetStorageImpl())) +
-              ".storage");
+                  tensor.storage().unsafeGetStorageImpl()));
```

Comment:
I guess the only caveat of using StorageImpl is that the id may change from run to run. Should not be a real issue, but for metadata comparison on data.pkl or bytecode.pkl, the tensor metadata may be different literally, although the tensor content does not change.

Reply:
Yes, the loss of determinism in the naming is not the best. Others were wanting this to be fixed as well. I can address it in a follow up PR